tests/linalg_qr_tests.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright © 2023 Apple Inc.
+
+#include "doctest/doctest.h"
+
+#include <iostream>
+#include "mlx/mlx.h"
+
+using namespace mlx::core;
+
+TEST_CASE("test QR factorization") {
+  array A = array({{2., 1., 1., 2.}, {2, 2}});
+  array out = linalg::qrf(A, default_stream(Device::cpu));
+  out.eval();
+
+  // std::cout << out << "\n";
+  CHECK_EQ(true, true);
+}
+
+// torch
+// Q=tensor([[-0.8944, -0.4472],
+//         [-0.4472,  0.8944]]),
+// R=tensor([[-2.2361, -1.7889],
+//         [ 0.0000,  1.3416]]))
+
+// mlx
+// 1: Q:array([[-0.894427, -0.447214],
+// 1:        [-0.447214, 0.894427]], dtype=float32)
+// 1: R:array([[-2.23607, 0.236068],
+// 1:        [0, 1.34164]], dtype=float32)
```

Comment:
you can see there's a slight mismatch in R matrix. the issue is mainly because lapack accepts col-major layout and mlx arrays are row-major. if I try transpose A before calling `triu`, i get the right output. however, i know transposing isn't ideal as mentioned in another thread.

i don't see any other way than manually re-arranging the memory layout to col-major and near the output change it back to row-major. ideas? @awni

Reply:
cc @awni ideas?